### PR TITLE
Fix pinning of packages not properly handled by the resolver

### LINKED ISSUE #2294: Cannot pin NETStandard.Library = 1.6.0
### Description

Similar to #2289, I seeming cannot install packages when I pin `NETStandard.Library 1.6.0`. This is required by AWS Lambda, but most recent `Microsoft.*` packages require version `1.6.1`. I keep getting 
the `Paket may still find a valid resolution, but this might take a while` message but I never waited long enough for it to finish.

Here's my intended dependencies file.

```
source https://www.nuget.org/api/v2/

nuget xunit >= 2.3 prerelease
nuget FakeItEasy
nuget FluentAssertions
nuget Microsoft.NET.Test.Sdk
nuget xunit.extensibility.execution >= 2.3 prerelease
nuget xunit.runner.visualstudio >= 2.3 prerelease
nuget Microsoft.AspNetCore.Mvc.Core
nuget Amazon.Lambda.Core
nuget Amazon.Lambda.Serialization.Json
nuget Amazon.Lambda.APIGatewayEvents
nuget Microsoft.AspNetCore.Server.Kestrel
nuget Microsoft.Extensions.Configuration.EnvironmentVariables
nuget Microsoft.Extensions.Configuration.FileExtensions
nuget Microsoft.Extensions.Configuration.Json
nuget Microsoft.Extensions.Logging
nuget Microsoft.Extensions.Options.ConfigurationExtensions
nuget AWSSDK.S3
nuget AWSSDK.Extensions.NETCore.Setup
nuget Amazon.Lambda.Logging.AspNetCore
nuget Amazon.Lambda.AspNetCoreServer
nuget NETStandard.Library 1.6.0
```

### Repro steps

Use the above dependencies file and `paket install`

### Expected behavior

Should resolve older versions of packages which don't require `NETStandard.Library: >= 1.6.1`

### Actual behavior

It takes very long

### Known workarounds

It is seemingly possible to pin all packages and their transitive dependencies but there are loads of them.

I also tried with groups but it keeps telling me that install fails because my project references two version of the `NETStandard.Library` package.

### PR DESCRIPTION
**Pull request by matthid**
I got started by the test of @0x53A, therefore this was done on top of his branch and contains https://github.com/fsprojects/Paket/pull/2305.

Fixes #2294 

Edit: Its actually OK to have this in the same PR because #2294 was the underlying issue the runtime solution didn't work out. So this actually first allows the workaround to "work" by applying the correct settings to the runtime resolution. At the same time it fixes #2294 itself (therefore a separate PR)

### CODE REVIEW

File: src/Paket.Core/PaketConfigFiles/DependenciesFile.fs [hunk 1]
```diff
@@ -229,6 +229,7 @@ type DependenciesFile(fileName,groups:Map<GroupName,DependenciesGroup>, textRepr
                 RemoteDownload.DownloadSourceFiles(Path.GetDirectoryName fileName, groupName, force, remoteFiles)
 
             // Step 1 Package resolution
```
Comment (cloudRoutine) [inline src/Paket.Core/PaketConfigFiles/DependenciesFile.fs]: let's call this something other than `step 1` & `step 2`, it's already confusing enough with all the other steps

Comment (matthid) [inline src/Paket.Core/PaketConfigFiles/DependenciesFile.fs]: Suggestions?

File: src/Paket.Core/PaketConfigFiles/DependenciesFile.fs [hunk 2]
```diff
@@ -262,17 +264,79 @@ type DependenciesFile(fileName,groups:Map<GroupName,DependenciesGroup>, textRepr
...
+                                | None -> group.Options.ResolverStrategyForTransitives
+                              Parent = PackageRequirementSource.DependenciesFile "runtimeresolution.dependencies"
+                              Graph = []
+                              Sources = group.Sources
+                              Settings =
+                                match oldDepsInfo with
+                                | Some d -> d.Settings
+                                | None -> p.Settings })
+                        |> fun des -> Seq.append des runtimeDeps
+                        |> Seq.distinctBy (fun p -> p.Name) |> Set.ofSeq
```
Comment (matthid) [inline src/Paket.Core/PaketConfigFiles/DependenciesFile.fs]: This part seems to be quite slow as it takes quite a while after `Trying to find a valid resolution considering runtime dependencies...` has been printed until the second resolution starts...

File: src/Paket/Paket.fsproj [hunk 3]
```diff
@@ -34,8 +34,8 @@
     <StartWorkingDirectory>D:\temp\coreclrtest</StartWorkingDirectory>
     <StartArguments>install</StartArguments>
     <StartWorkingDirectory>C:\dev\src\Paket\integrationtests\scenarios\loading-scripts\dependencies-file-flag\temp</StartWorkingDirectory>
-    <StartArguments>install</StartArguments>
-    <StartWorkingDirectory>C:\PROJ\Paket\integrationtests\scenarios\i001145-excludes\temp</StartWorkingDirectory>
+    <StartArguments>update</StartArguments>
+    <StartWorkingDirectory>C:\PROJ\Paket\integrationtests\scenarios\i002294-pin-netstandard-1-6\temp</StartWorkingDirectory>
```
Comment (cloudRoutine) [inline src/Paket/Paket.fsproj]: remove the detritus

**Comment by cloudRoutine** [comment]
aside from some superficial cleanup LGTM

File: src/Paket.Core/Dependencies/PackageResolver.fs [hunk 4]
```diff
@@ -693,7 +713,7 @@ let Resolve (getVersionsF, getPackageDetailsF, groupName:GroupName, globalStrate
                         currentStep.CurrentResolution.Count 
                         (currentStep.CurrentResolution |> Seq.map (fun x -> sprintf "\n     - %O, %O" x.Key x.Value.Version) |> String.Concat)
                         currentStep.OpenRequirements.Count
-                        (currentStep.OpenRequirements  |> Seq.map (fun x -> sprintf "\n     - %O, %O" x.Parent x.VersionRequirement) |> String.Concat)
+                        (currentStep.OpenRequirements  |> Seq.map (fun x -> sprintf "\n     - %O, %O (from %O)" x.Name x.VersionRequirement x.Parent) |> String.Concat)
```
Comment (matthid) [inline src/Paket.Core/Dependencies/PackageResolver.fs]: @cloudRoutine what about this?

**Pull request merged**